component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -755,6 +758,15 @@ private void redirectToMobileNoReqPage(HttpServletResponse response, Authenticat
                 String mobileNumberPatternViolationError = SMSOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_VIOLATED;
                 String mobileNumberPattern =
                         context.getAuthenticatorProperties().get(SMSOTPConstants.MOBILE_NUMBER_REGEX);
+                if (isMobileNumberUpdateFailed(context)) {
+                    url = url + SMSOTPConstants.RETRY_PARAMS;
```

Comment:
use FrameworkUtils.appendQueryParamStringtoURL

Reply:
Fixed this issue